java/src/main/java/com/logicalclocks/hsfs/FeatureGroup.java

```diff
@@ -260,49 +418,303 @@ public void save(Dataset<Row> featureData, Map<String, String> writeOptions)
+   *
+   * <pre>
+   * {@code
+   *        // get feature store handle
+   *        FeatureStore fs = ...;
+   *        // get feature group handle
+   *        FeatureGroup fg = ...;
+   *        //insert feature data
+   *        fg.insert(featureData);
+   * }
+   * </pre>
+   *
+   * @param featureData spark DataFrame, RDD. Features to be saved.
+   * @throws IOException Generic IO exception.
+   * @throws FeatureStoreException If client is not connected to Hopsworks; cannot run read query on storage and/or
+   *                               can't reconcile HUDI schema.
+   * @throws ParseException In case it's unable to parse HUDI commit date string to date type.
+   */
   public void insert(Dataset<Row> featureData) throws IOException, FeatureStoreException, ParseException {
     insert(featureData, null, false);
   }
 
+  /**
+   * Persist the metadata and materialize the feature group to the feature store or insert data from a dataframe into
+   * the existing feature group.
+   * Incrementally insert data to a feature group or overwrite all  data contained in the feature group.
+   * The `features` dataframe can be a Spark DataFrame or RDD.
+   * If the feature group doesn't exist, the insert method will create the necessary metadata the first time it is
+   * invoked and write the specified `features` dataframe as feature group to the online/offline feature store.
+   *
+   * <pre>
+   * {@code
+   *        // get feature store handle
+   *        FeatureStore fs = ...;
+   *        // get feature group handle
+   *        FeatureGroup fg = ...;
+   *        // Define additional write options.
+   *        Map<String, String> writeOptions = ...;
+   *        // insert feature data
+   *        fg.insert(featureData, writeOptions);
+   * }
+   * </pre>
+   *
+   * @param featureData Spark DataFrame, RDD. Features to be saved.
+   * @param writeOptions Additional write options as key-value pairs.
+   * @throws IOException Generic IO exception.
+   * @throws FeatureStoreException If client is not connected to Hopsworks; cannot run read query on storage and/or
+   *                               can't reconcile HUDI schema.
+   * @throws ParseException In case it's unable to parse HUDI commit date string to date type.
+   */
   public void insert(Dataset<Row> featureData,  Map<String, String> writeOptions)
       throws FeatureStoreException, IOException, ParseException {
     insert(featureData, null, false, null, writeOptions);
   }
 
+  /**
+   * Persist the metadata and materialize the feature group to the feature store or insert data from a dataframe into
+   * the existing feature group.
+   * Incrementally insert data to a feature group or overwrite all  data contained in the feature group. By default,
+   * the data is inserted into the offline storage as well as the online storage if the feature group is
+   * `onlineEnabled=true`. To insert only into the online storage, set `storage="online"`, or oppositely
```

Comment:
This is wrong, it's not Python :) The code snippet is correct, you need the enum to specify which storage to write to, not the string.